homeassistant/components/enphase_envoy/coordinator.py

```diff
@@ -144,7 +144,11 @@ async def _async_update_data(self) -> dict[str, Any]:
                 if not self._setup_complete:
                     await self._async_setup_and_authenticate()
                     self._async_mark_setup_complete()
-                return (await envoy.update()).raw
+                # dump all received data in debug mode to assist troubleshooting
+                envoy_data = await envoy.update()
+                assert envoy_data is not None
```

Comment:
we should not use assert

Reply:
Removed it, thanks @dgomes

Reply:
no problem, but you should now test that its not `None`

Reply:
Ok, but wouldn't the debug statement not just print None in the log? 

The original code just returned .raw without test. So I guess the test needs to be for None and return None or if not None return Raw?

Reply:
`envoy_data.raw`   None has no raw property :)

Reply:
Ah I see. Considering the original `return (await envoy.update()).raw`  I assumed it was not an issue.

The update method returns class EnvoyData from pyenphase. Some or all members may be None, but the class itself not I guess.

```
    async def update(self) -> EnvoyData:
```

```
@dataclass(slots=True)
class EnvoyData:
    """Model for an envoy."""

    encharge_inventory: dict[str, EnvoyEncharge] | None = None
    encharge_power: dict[str, EnvoyEnchargePower] | None = None
.
.
.
    raw: dict[str, Any] = field(default_factory=dict)
```

Thanks for explaining, pretty new to python (clearly showing).